File: claude/Dockerfile.js
FROM docker/sandbox-templates:claude-code

USER root

# ── System deps ──────────────────────────────────────────
RUN apt-get update && apt-get install -y --no-install-recommends \
    build-essential gcc g++ make cmake \
    pkg-config libssl-dev libffi-dev zlib1g-dev \
    python3 python3-dev \
    curl wget unzip htop tree strace \
    && rm -rf /var/lib/apt/lists/*

# ── Node.js 22 LTS via NodeSource ───────────────────────
# Base image has Node but we ensure latest LTS
RUN curl -fsSL https://deb.nodesource.com/setup_22.x | bash - && \
    apt-get install -y --no-install-recommends nodejs && \
    rm -rf /var/lib/apt/lists/*

# ── Bun ──────────────────────────────────────────────────
RUN curl -fsSL https://bun.sh/install | bash && \
    cp /root/.bun/bin/bun /usr/local/bin/bun && \
    cp /root/.bun/bin/bunx /usr/local/bin/bunx

# ── pnpm ─────────────────────────────────────────────────
RUN npm install -g pnpm

# ── Global dev tools ────────────────────────────────────
RUN npm install -g \
    typescript \
    tsx \
    ts-node \
    @types/node \
    eslint \
    prettier \
    vitest \
    jest \
    nodemon \
    concurrently \
    turbo \
    wrangler \
    vercel \
    @biomejs/biome

# ── Framework CLIs ──────────────────────────────────────
RUN npm install -g \
    next \
    create-next-app \
    @nestjs/cli \
    prisma \
    drizzle-kit \
    @anthropic-ai/sdk \
    openai

# ── Ensure agent user has access ────────────────────────
RUN mkdir -p /home/agent/.npm-global && \
    chown -R agent:agent /home/agent/.npm-global
ENV NPM_CONFIG_PREFIX="/home/agent/.npm-global"
ENV PATH="/home/agent/.npm-global/bin:${PATH}"

USER agent

RUN echo 'export PATH="$HOME/.npm-global/bin:$PATH"' >> /home/agent/.bashrc

USER root
RUN claude update --yes 2>/dev/null || true
USER agent

File: claude/Dockerfile.cpp
FROM docker/sandbox-templates:claude-code

USER root

# ── Core C++ toolchain + libraries (single apt layer) ───
RUN apt-get update && apt-get install -y --no-install-recommends \
    build-essential gcc g++ gcc-13 g++-13 \
    clang clang-format clang-tidy clang-tools \
    lld lldb libc++-dev libc++abi-dev \
    gdb valgrind cmake make ninja-build \
    pkg-config autoconf automake libtool \
    zip unzip tar \
    libssl-dev libffi-dev zlib1g-dev \
    libboost-dev \
    libbenchmark-dev libgtest-dev \
    libfmt-dev nlohmann-json3-dev \
    libspdlog-dev libabsl-dev \
    libprotobuf-dev protobuf-compiler \
    libgrpc++-dev \
    strace ltrace binutils elfutils \
    cppcheck iwyu \
    doxygen graphviz \
    htop tree curl wget \
    && rm -rf /var/lib/apt/lists/*

# ── vcpkg (C++ package manager) ─────────────────────────
RUN git clone --depth 1 https://github.com/microsoft/vcpkg.git /opt/vcpkg && \
    /opt/vcpkg/bootstrap-vcpkg.sh && \
    ln -s /opt/vcpkg/vcpkg /usr/local/bin/vcpkg
ENV VCPKG_ROOT="/opt/vcpkg"

# ── Conan (C++ package manager) ─────────────────────────
RUN pip3 install --break-system-packages conan

# ── Set default to gcc-13 ───────────────────────────────
RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-13 130 \
    --slave /usr/bin/g++ g++ /usr/bin/g++-13

USER agent

USER root
RUN claude update --yes 2>/dev/null || true
USER agent
